src/framework/loader.ts

```diff
@@ -1,32 +1,23 @@
 import { GroupRecorder } from './logger.js';
 import { ParamsAny, paramsEquals, paramsSupersets } from './params/index.js';
-import { RunCaseIterable, TestCaseID, RunCase } from './test_group.js';
+import { RunCaseIterable, RunCase } from './test_group.js';
 import { allowedTestNameCharacters } from './allowed_characters.js';
+import { TestSuiteListing } from './listing.js';
+import { TestSpecID, TestCaseID } from './id.js';
 
-export interface TestGroupDesc {
-  readonly path: string;
-  readonly description: string;
-}
-
-export interface TestSuiteListing {
-  readonly suite: string;
-  readonly groups: Iterable<TestGroupDesc>;
-}
-
+// One of the following:
+// - A shell object describing a directory (from its README.txt).
```

Comment:
It's weird that the README would be there, can we use a sum type with README | TestSpecFile wherever this is needed?